akka-docs/rst/java/stream/stream-integrations.rst

```diff
@@ -7,138 +7,109 @@ Integration
+Messages can be sent to a stream with ``Source.queue`` or via the :class:`ActorRef` that is
+materialized by ``Source.actorRef``.
 
-The actor will be stopped when the stream is completed, failed or cancelled from downstream,
-i.e. you can watch it to get notified when that happens.
+mapAsync + ask
+^^^^^^^^^^^^^^
 
-Sink.actorRef
-^^^^^^^^^^^^^
+A nice way to delegate some processing of elements in a stream to an actor is to 
+use ``ask`` in ``mapAsync``. The back-pressure of the stream is maintained by
+the ``CompletionStage`` of the ``ask`` and the mailbox of the actor will not be filled with
+more messages than the given ``parallelism`` of the ``mapAsync`` stage.
 
-The sink sends the elements of the stream to the given `ActorRef`. If the target actor terminates
-the stream will be cancelled. When the stream is completed successfully the given ``onCompleteMessage``
-will be sent to the destination actor. When the stream is completed with failure a ``akka.actor.Status.Failure``
-message will be sent to the destination actor.
+.. includecode:: ../code/docs/stream/IntegrationDocTest.java#mapAsync-ask
 
-.. warning::
+Note that the messages received in the actor will be in the same order as
+the stream elements, i.e. the ``parallelism`` does not change the ordering
+of the messages. There is a performance advantage of using parallelism > 1
+even though the actor will only process one message at a time because then there
+is already a message in the mailbox when the actor has completed previous
+message. 
 
-   There is no back-pressure signal from the destination actor, i.e. if the actor is not consuming
-   the messages fast enough the mailbox of the actor will grow. For potentially slow consumer actors
-   it is recommended to use a bounded mailbox with zero `mailbox-push-timeout-time` or use a rate
-   limiting stage in front of this stage.
+The actor must reply to the ``getSender()`` for each message from the stream. That
+reply will complete the ``CompletionStage`` of the ``ask`` and it will be the element that
+is emitted downstreams from ``mapAsync``.
 
-ActorPublisher
-^^^^^^^^^^^^^^
+.. includecode:: ../code/docs/stream/IntegrationDocTest.java#ask-actor
 
-Extend :class:`akka.stream.actor.AbstractActorPublisher` to implement a
-stream publisher that keeps track of the subscription life cycle and requested elements.
+The stream can be completed with failure by sending ``akka.actor.Status.Failure``
+as reply from the actor.
 
-Here is an example of such an actor. It dispatches incoming jobs to the attached subscriber:
+If the ``ask`` fails due to timeout the stream will be completed with
+``TimeoutException`` failure. If that is not desired outcome you can use ``recover`` 
+on the ``ask`` :class:`CompletionStage`.
 
-.. includecode:: ../code/docs/stream/ActorPublisherDocTest.java#job-manager
+If you don't care about the replies you can use ``Sink.ignore`` after the ``mapAsync``
+stage and then actor is effectively a sink of the stream.
```

Comment:
Maybe clarify that it is when you don't care but still want request-response to backpressure? I first read this and thought "why would I use ask if I don't care about the reply?".